hnn_core/tests/test_dipole.py

```diff
@@ -53,3 +53,5 @@ def test_num_trials():
     net = Network(params)
     with pytest.raises(ValueError, match="Invalid number of simulations: 0"):
         simulate_dipole(net, n_trials=0)
+    with pytest.raises(TypeError, match="record_vsoma must be bool, got int"):
+        simulate_dipole(net, n_trials=1, record_vsoma=0)
```

Comment:
perhaps also a check that `net.vsoma` is `None` or the attribute is missing when `record_vsoma=False` ?

Reply:
So there's two paths with this. The first would be enabling voltage recording in the `simulate_dipole()` call of `test_parallel_backends.py`. The second would be running a separate simulation for voltage recording. My preference is for the first as this keeps the test time short.

In either case enabling voltage recordings during the tests will likely reintroduce the unresolved MPI errors. It's not ideal to merge this PR without full tests, but their introduction will require #198 to be merged.

Reply:
> My preference is for the first as this keeps the test time short.

you can still use a tiny network with a small `tstop` so it runs super-fast. We don't check the voltage traces, just that the dimensions and object behave as we want.

Reply:
> enabling voltage recordings during the tests will likely reintroduce the unresolved MPI errors.

just do it for the joblibs backend. We can improve it in later PRs but having a placeholder test will help.